hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/SignedStateCommand.java

```diff
@@ -19,16 +19,134 @@
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.config.Configurator;
 import picocli.CommandLine;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParentCommand;
 
 /**
  * A subcommand of the {@link PlatformCli}, for dealing with signed state files
  */
+@SuppressWarnings("java:S106") // "use of system.out/system.err instead of logger" - not needed/desirable for CLI tool
 @CommandLine.Command(
         name = "signed-state",
         mixinStandardHelpOptions = true,
         description = "Operations on signed-state files.")
 @SubcommandOf(PlatformCli.class)
 public final class SignedStateCommand extends AbstractCommand {
+
+    @ParentCommand
+    PlatformCli parent;
+
+    enum Verbosity {
+        SILENT,
+        VERBOSE
+    }
+
+    @Option(
+            names = {"-f", "--file"},
+            arity = "1",
+            description = "Input signed state file")
+    Path inputFile;
+
+    @Option(
+            names = {"-c", "--config"},
+            description = "A path to where a configuration file can be found. If not provided then defaults are used.")
+    private void setConfigurationPath(@NonNull final List<Path> configurationPaths) {
+        Objects.requireNonNull(configurationPaths, "configurationPaths");
+
+        configurationPaths.forEach(this::pathMustExist);
+        this.configurationPaths = configurationPaths;
+    }
+
+    List<Path> configurationPaths = List.of();
+
+    @Option(
+            names = {"-v", "--verbose"},
+            arity = "0..1",
+            defaultValue = "false",
+            description = "Verbosity of command")
+    private void setVerbosity(final boolean doVerbose) {
+        verbosity = doVerbose ? Verbosity.VERBOSE : Verbosity.SILENT;
+    }
+
+    Verbosity verbosity;
+
+    @Option(
+            names = {"--log-level"},
+            arity = "1",
+            converter = LogLevelConverter.class,
+            defaultValue = "OFF",
+            description = "Log4j level for root logger (can be one of ${COMPLETION-CANDIDATES})")
+    private void setRootLogLevel(final Level loggingLevel) {
+        this.loggingLevel = loggingLevel;
+    }
+
+    Level loggingLevel;
+
+    @SuppressWarnings("java:S4792") // "make sure this logger's configuration is safe"
+    private void setRootLoggingLevel() {
+        // BUG: This doesn't work.  Setting level to `WARN` I still see a message from `INFO STATE_TO_DISK`
+        if (verbosity == Verbosity.VERBOSE) System.out.printf("===Log level set to %s%n", loggingLevel);
+        var logger = LogManager.getRootLogger();
+        Configurator.setAllLevels(logger.getName(), loggingLevel);
+        logger = LogManager.getLogger(FileUtils.class);
+        Configurator.setAllLevels(logger.getName(), loggingLevel);
```

Comment:
Interesting concept, but I'm not surprised this didn't work. Other loggers can easily override the root logger by design

Reply:
It _used_ to work. But doesn't now.  Guess it's fragile.  I'll probably rip it out, thank you for the impetus.  But what I _wanted_ to do was make this a kind of _clean_ command line tool and not get it interrupted with all the logging cruft.  The _root cause_ here is that I don't understand how to _configure_ log4j properly, so that's all on me.

Reply:
(Plus I have to look up what the "root logger" is for log4j if it can be overridden, must not mean what I thought it sounds like it means.)

Reply:
Totally fine to do this in a subsequent PR 👍 

I've always understood the root logger to be the default, per se, when other loggers aren't configured on top of it. For example, if the root logger is configured at level `WARN` but I have a logger configured at level `INFO` for the `com.swirlds` package, I would expect any class in the `com.swirlds` package to log at the `INFO` level. Maybe I've misunderstood this all these years (wouldn't surprise me).

In this case I suspect that some other logger is getting included. This solution should be fine, it's just not complete. You can programmatically filter out other loggers (though I'd have to search a bit to figure out how, I usually use the `log4j2.xml` file). Happy to look if it'd be helpful though

Reply:
done